CMSgov/cms-carts-seds
# remove momentjs

### Description

Replace momentjs with `date-fns` and native Date type

the moment package is a sub-dependency of `serverless-bundle`. we cannot remove it from yarn.lock, but hopefully this still resolves the issue.

References:
[Security issue](https://www.invicti.com/web-vulnerability-scanner/vulnerabilities/version-disclosure-momentjs/)
[MomentJS recommends using alternatives](https://momentjs.com/docs/#/-project-status/)
[date-fns docs](https://date-fns.org/v3.6.0/docs/Getting-Started)

### Related ticket(s)


CMDCT-3906

---
### How to test

Tests pass

Timeout
- Modify the values for `IDLE_WINDOW` and `PROMPT_AT` in the `services/ui-src/src/hooks/authHooks/authLifecycle.js` file to be shorter (change leading values to 1 and 0.5 or something)
- Run locally
- Let the app sit until the timeout hits
  - Ensure the prompt shows up as expected
  - Ensure the prompt still does the correct actions when you click on them
  - Ensure if you don't do anything it logs you out

Save message
- Enter any report as a state user and change something
- Verify the autosave banner switches from "Not yet saved" to "Saved"
- Verify that if you let it sit for 1+ minutes the message changes to something like "Last saved 9:30 AM (5 minutes ago)"

Thanks message
- Enter any report as a state user and submit it
- Verify the message shows the date and time formatted correctly
  - "Submitted on August 28th, 2024 at 9:35 AM by Frank States."
- Verify that if you refresh the page it still shows the correct info

Page info
- Enter any report as a state user
- Verify that if you make a change the section at the top that says "Draft | Last Edited ... " shows the correct date

### Notes


Our SaveMessage had cases for showing a message if the last save was multiple days or years ago, but our auth times out after 30 minutes. I have doubts that our state users would stay up overnight to stay logged in, so I think it's safe to remove those cases.

---
### Pre-review checklist
